Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/TransformComponent.cpp

```diff
@@ -860,17 +860,16 @@ namespace AzToolsFramework
                     return;
                 }
 
-                bool isDuringUndoRedo = false;
-                EBUS_EVENT_RESULT(isDuringUndoRedo, AzToolsFramework::ToolsApplicationRequests::Bus, IsDuringUndoRedo);
-                if (!isDuringUndoRedo)
+                if (!m_initialized)
                 {
-                    // When parent comes online, compute local TM from world TM.
-                    CheckApplyCachedWorldTransform(parentTransform->GetWorldTM());
+                    m_initialized = true;
+                    // OnTransformChanged should only be called once when entity is recreated
+                    OnTransformChanged(AZ::Transform::Identity(), parentTransform->GetWorldTM());
                 }
                 else
                 {
-                    // During undo operations, just apply our local TM.
-                    OnTransformChanged(AZ::Transform::Identity(), parentTransform->GetWorldTM());
+                    // When parent comes online, compute local TM from world TM.
```

Comment:
nit : m_initialized and 'parent coming online' is not an obvious 1-1 mapping. I suggest changing this comment to say something like 'If this is not the first time this entity is being activated, compute...'